src/dataaggregator/aggregator/aggregator_test.go

```diff
@@ -97,55 +97,95 @@ var _ = Describe("Aggregator", func() {
-						Threshold:          30,
-						Operator:           "<",
-						Adjustment:         "-1",
-					}}},
-			}}
+
+		})
+		Context("when there are data in triggerMap", func() {
+			JustBeforeEach(func() {
+				triggerMap = map[string]*Trigger{testAppId: &Trigger{
+					AppId: testAppId,
+					TriggerRecord: &TriggerRecord{
+						InstanceMaxCount: 5,
+						InstanceMinCount: 1,
+						ScalingRules: []*ScalingRule{&ScalingRule{
+							MetricType:       "MemoryUsage",
+							StatWindow:       300,
+							BreachDuration:   300,
+							CoolDownDuration: 300,
+							Threshold:        30,
+							Operator:         "<",
+							Adjustment:       "-1",
+						}}},
+				}}
+			})
+			It("should parse the triggers to appmonitor and put them in appChan", func() {
+				aggregator.ConsumeTrigger(triggerMap, appChan)
+				Eventually(appChan).Should(Receive(&appMonitor))
+				Expect(appMonitor).To(Equal(&AppMonitor{
+					AppId:      testAppId,
+					MetricType: "MemoryUsage",
+					StatWindow: 300,
+				}))
+			})
 		})
-		It("should parse the triggers to appmonitor and put them in appChan", func() {
-			aggregator.ConsumeTrigger(triggerMap, appChan)
-			Eventually(appChan).Should(Receive(&appMonitor))
-			Expect(appMonitor).To(Equal(&AppMonitor{
-				AppId:          testAppId,
-				MetricType:     "MemoryUsage",
-				StatWindowSecs: 300,
-			}))
+		Context("when there is not data in triggerMap", func() {
+			JustBeforeEach(func() {
+				triggerMap = map[string]*Trigger{}
+			})
+			It("should not receive any data from the appChan", func() {
+				aggregator.ConsumeTrigger(triggerMap, appChan)
+				Consistently(appChan).ShouldNot(Receive())
+			})
 		})
+		Context("when the triggerMap is nil", func() {
+			JustBeforeEach(func() {
+				triggerMap = nil
+			})
+			It("should not receive any data from the appChan", func() {
+				aggregator.ConsumeTrigger(triggerMap, appChan)
+				Consistently(appChan).ShouldNot(Receive())
+			})
+		})
+
 	})
 	Context("ConsumeAppMetric", func() {
 		var appmetric *AppMetric
 		JustBeforeEach(func() {
 			aggregator = NewAggregator(logger, clock, TestPolicyPollerInterval, database, metricServer.URL(), TestMetricPollerCount)
-			appmetric = &AppMetric{
-				AppId:      testAppId,
-				MetricType: metricType,
-				Value:      250,
-				Unit:       "bytes",
-				Timestamp:  timestamp}
+
 		})
-		It("should save the appmetric to database", func() {
-			aggregator.ConsumeAppMetric(appmetric)
-			Eventually(database.SaveAppMetricCallCount).Should(BeNumerically("==", 1))
+		Context("when there is data in appmetric", func() {
+			JustBeforeEach(func() {
+				appmetric = &AppMetric{
+					AppId:      testAppId,
+					MetricType: metricType,
+					Value:      250,
+					Unit:       "bytes",
+					Timestamp:  timestamp}
+			})
+			It("should call database.SaveAppmetric to save the appmetric to database", func() {
+				aggregator.ConsumeAppMetric(appmetric)
+				Eventually(database.SaveAppMetricCallCount).Should(BeNumerically("==", 1))
```

Comment:
Should(Equal(1))
BeNumerically should be used when you need > or >= like tests.